scripts/recovery_sync

```diff
@@ -24,6 +24,17 @@ target="$1"
 workdir=$(mktemp -d)
 trap cleanup EXIT
 
+#
+# When syncing the recovery environment, we want to ensure that only a single
+# process is attempting to do this at any given time.  Thus, we add some
+# locking in case multiple processes call this script concurrently; to prevent
+# accidental corruption and/or failures.
+#
+if [[ "${SYNC_EXECUTE_LOCKED:-x}" != "true" ]]; then
```

Comment:
Why `"${SYNC_EXECUTE_LOCKED:-x}"` instead of just `"$SYNC_EXECUTE_LOCKED"`?

Also perhaps call this RECOVERY_SYNC_LOCKED? In the example Prakash gave he used UPGRADE_EXECUTE_LOCKED because it's the "upgrade/execute".

Reply:
Because we have `set -eux` on, an attempt to use an unset variable will cause the script to fail. Using :+ allows us to provide a default value if the variable is unset.